servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultRequestTracker.java

```diff
@@ -61,26 +61,29 @@ final class DefaultLatencyTracker implements LatencyTracker {
     private int ewma;
     private volatile int pending;
 
-    DefaultLatencyTracker(final long halfLifeNanos, final LongSupplier currentTimeSupplier) {
-        this(halfLifeNanos, currentTimeSupplier, DEFAULT_CANCEL_PENALTY, DEFAULT_ERROR_PENALTY);
+    DefaultRequestTracker(final long halfLifeNanos) {
+        this(halfLifeNanos, DEFAULT_CANCEL_PENALTY, DEFAULT_ERROR_PENALTY);
     }
 
-    DefaultLatencyTracker(final long halfLifeNanos, final LongSupplier currentTimeSupplier,
-                          long cancelPenaly, long errorPenaly) {
+    DefaultRequestTracker(final long halfLifeNanos, final long cancelPenalty, final long errorPenalty) {
         if (halfLifeNanos <= 0) {
             throw new IllegalArgumentException("halfLifeNanos: " + halfLifeNanos + " (expected >0)");
         }
         this.invTau = Math.pow((halfLifeNanos / log(2)), -1);
-        this.currentTimeSupplier = currentTimeSupplier;
-        this.lastTimeNanos = currentTimeSupplier.getAsLong();
-        this.cancelPenalty = cancelPenaly;
-        this.errorPenalty = errorPenaly;
+        this.cancelPenalty = cancelPenalty;
+        this.errorPenalty = errorPenalty;
     }
 
+    /**
+     * The current time in nanoseconds.
+     * @return the current time in nanoseconds.
+     */
+    protected abstract long currentTimeNanos();
```

Comment:
If the `DefaultRequestTracker` can access `Executor` from `HealthCheckerFactory.newHealthChecker(...)`, I would recommend using `Executor` as a `TimeSource` instead of adding an internal hook here, as a consistent approach we use in other places.